### PR TITLE
Correctly handle empty arguments when creating processes

### PR DESCRIPTION
**Pull request by OneBlue**
## Summary of the Pull Request

This change updates the init protocol to correctly handle empty strings in array. 

This change also test coverage for empty arguments at the VM, container and exec level

## PR Checklist

- [ ] **Closes:** Link to issue #xxx
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [X] **Tests:** Added/updated if needed and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated if needed
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/wsl/) and link it here: #xxx

### CODE REVIEW

File: src/shared/inc/message.h [hunk 1]
```diff
@@ -147,28 +147,34 @@ class MessageWriter
...
-            totalSize += size + 1;
+            totalSize += strlen(String[i]) + sizeof(int32_t);
         }
 
         auto span = InsertBuffer(Index, totalSize);
-
         auto it = span.begin();
+
+        auto insertSize = [&](int32_t size) {
+            it = std::copy(reinterpret_cast<const std::byte*>(&size), reinterpret_cast<const std::byte*>(&size) + sizeof(size), it);
```
Comment (benhillis) [inline src/shared/inc/message.h:164]: wouldn't using gsl::copy be better here?

Comment (OneBlue) [reply src/shared/inc/message.h:164]: I don't think so, because gsl::copy doesn't return the end iterator, so we have to manually increase it after it returns

**Review by benhillis — APPROVED** [review]
